### PR TITLE
[WIP]Associate modal on host group list And Launch Modal on Template Lists persists on Sort/Pagination

### LINKED ISSUE #10181: Host Groups Associate Modals Closes on Sort
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
The Associate modal closes when you sort the Host Groups names.

##### STEPS TO REPRODUCE

Click on Hosts --> Groups Tab --> Associate ---> Sort by name

##### EXPECTED RESULTS

Group names are returned in the modal in reverse alpha order

##### ACTUAL RESULTS

Modal Closes

##### ADDITIONAL INFORMATION
modalcloses

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
Resolved #10181 and https://github.com/ansible/awx/issues/9218

E2E test started

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.test.jsx [hunk 1]
```diff
@@ -280,4 +280,41 @@ describe('<HostGroupsList />', () => {
     wrapper.update();
     expect(wrapper.find('AlertModal ErrorDetail').length).toBe(1);
   });
+
+  test.only('Associate modal does not close on sort', async () => {
```
Comment (nixocio) [inline awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.test.jsx]:
```suggestion
  test('Associate modal does not close on sort', async () => {
```

**Comment by jakemcdermott** [comment]
@AlexSCorey @nixocio 
The bug addressed by this PR reminds me of https://github.com/ansible/awx/issues/9218. After reading through this PR, I'm now wondering if it can be solved in a similar fashion. Thoughts?

**Comment by AlexSCorey** [comment]
@jakemcdermott I think you are right. I can add that fix to this PR as well

File: awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.test.jsx [hunk 2]
```diff
@@ -280,4 +280,41 @@ describe('<HostGroupsList />', () => {
...
   });
+
+  test('Associate modal does not close on sort', async () => {
+    InventoriesAPI.readGroups.mockResolvedValue({
+      data: {
+        results: [
+          { id: 1, name: 'Group A', url: '' },
+          { id: 2, name: 'Group B', url: '' },
+        ],
+        count: 1,
```
Comment (nixocio) [inline awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.test.jsx:291]:
```suggestion
        count: 2,
```

File: awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.test.jsx [hunk 3]
```diff
@@ -280,4 +280,41 @@ describe('<HostGroupsList />', () => {
...
+        .find('Button[aria-label="Sort"]')
+        .prop('onClick')()
+    );
+    InventoriesAPI.readGroups.mockResolvedValue({
+      data: {
+        results: [
+          { id: 2, name: 'Group B', url: '' },
+          { id: 1, name: 'Group A', url: '' },
+        ],
+        count: 1,
```
Comment (nixocio) [inline awx/ui_next/src/screens/Host/HostGroups/HostGroupsList.test.jsx:311]:
```suggestion
        count: 2,
```

**Review by nixocio — APPROVED** [review]

**Comment by AlexSCorey** [comment]
this needs a bit more investigation before moving forward

**Comment by AlexSCorey** [comment]
Closing this PR after realizing that the issue is larger than originally thought. The plan is for https://github.com/ansible/awx/pull/10215 to fix the issues this PR attempted to fix.

**Pull request closed without merging**